pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -243,14 +223,14 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 		return mapCtx.Err()
 	}
 
-	tokens := strings.Split(created.Name, "/")
-	parent, err = a.id.Parent()
+	_, name, err := krm.ParseConnectionExternal(created.Name)
```

Comment:
Looking at this, I think it would read well if it was a method on BigQueryConnectionID, like this:

```
name := &BigQueryConnectionID{}
if err := name.ParseSelfLink(created.Name); err != nil {
....
```